src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,130 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (arg: ()=> Movie[]) => void;
+  listMovie: Movie[];
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd, listMovie }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setId] = useState('');
+  const [isDisable, setIsDisable] = useState(true);
+
+  useEffect(() => {
+    if (title && imgUrl && imdbUrl && imdbId) {
+      setIsDisable(false);
+    } else {
+      setIsDisable(true);
+    }
```

Comment:
```suggestion
    setIsDisable(title && imgUrl && imdbUrl && imdbId);
```
we can simplify it, but do we really need to add a separate state for `isDisable`, I suggest just using it as a variable